### ISSUE TITLE
Issue with the regex

### ISSUE THREAD

**Issue by Harish0596** [RasaHQ/rasa_nlu#950]
**Rasa NLU version** : 0.12.0a2

**Used backend / pipeline** : spacy_sklearn

**Operating system** : Windows 10

**Issue**: regex is not working
I have created a regex expression to identify a three digit number in markdown format which is my entity and also slot . After training it is successfully building the regex_featurizer but it is taking any n - digit  number into the slot.

**Content of configuration file** (if used & relevant):
```
## regex:mid
 - [1-9]{3}  
```

**Comment by amn41** [comment]
the pipeline you are using includes the `ner_crf` component, which looks at a number of features in the text to decide if there is an entity. The regex you provide is not treated as a rule, but just creates another feature. 

TLDR add something like this to your config:
```
  "ner_crf": {
    "BILOU_flag": false,
    "features": [
      ["low", "title", "upper", "pos", "pos2"],
      ["bias", "low", "word3", "word2", "upper", "title", "pos", "pos2", "pattern"],
      ["low", "title", "upper", "pos", "pos2"]],
    "max_iterations": 50,
    "L1_c": 1,
    "L2_c": 1e-3
  },
```

Here I removed the 'digit' feature so the the crf can only see your custom regex feature, and doesn't look for the default 'is there a digit in this token' feature.

**Comment by Harish0596** [comment]
even i tried removing the 'digit' after building the model, in this case it's not even detecting the slot value

**Comment by akelad** [comment]
Just to sanity check: have you labeled all these numbers in your NLU examples as entities?

**Comment by Harish0596** [comment]
@akelad I have given enough training data with numbers(3-digit numbers) as entities

**Comment by wrathagom** [comment]
@Harish0596 can you provide your training data?

**Comment by Harish0596** [comment]
```
## intent:order_details
- order details
- details of my order
- what about order with id [843](orderId)
- give the order details
- order details pls
- [832](orderId)
- give the details of the order with id [483](orderId)  
- i want to know my order details
- details of order with id [905](orderId)
- get the details following order with id [735](orderId)
- [984](orderId) order details
- can you give some details of order [429](orderId)
- pls get the details of order
- order id is [985](orderId)
- [913](orderId) is my order id
- ok then what about [563](orderId)
- wanna some info about my orders
- what about my orders
- orders
- order [183](orderId) details
- get order [127](orderId) information
- i want some info about [347](orderId) order
- i need details of my order
```

**Comment by akelad** [comment]
@Harish0596 You probably need some more training examples. However, a much easier solution is to use a pre-trained ner model like `ner_duckling` that automatically extracts numbers for you. You can then check in a custom action whether the number has exactly 3 digits and behave accordingly

**Comment by Harish0596** [comment]
@akelad  we can check easily with the custom action, but I want rasa to do that because I was trying to make use of all the features of Rasa
